oeps/best-practices/oep-0066-bp-authorization.rst

```diff
@@ -0,0 +1,460 @@
+(i.e. an implicit student role). 
+In this way, it can grant permissions to a user that they will not be able to use.
+
+auth_permission users and groups are assigned through the Django Admin Dashboard. Each 
+service can have its own Django Admin Dashboard. In the Open edX software, the LMS Django Admin Dashboard 
+will be used to control (most) user and group permissions.
+
+student_courseaccessrole
+------------------------
+
+Explicit roles are assigned to users, generally on a course level basis. 
+
+The roles are hardcoded strings that can be granted in the LMS or CMS.
+In addition to granting the roles in the UI, it is possible to assign 
+the roles through the LMS Django Admin Dashboard. 
+
+Each role assignment will generate one row in the database table. The values 
+in the row will determine if the user is granted access for a single course, all 
+courses in the org, or all courses in the instance.
+
+* If the course_id is not nil, the role grants permissions on the course level.
+* If the course_id is nil and the org_id is not nil, the role grants permissions on the organization level.
+* If the course_id and org_id are both nil, the role grants permissions on the instance level.
+
+django_comment_client_role
+--------------------------
+
+Explicit roles are assigned to users on a course level basis. 
+
+These roles require that the user already be enrolled in the course 
+(have an enrollment, audit or verified).
+
+Roles are assigned through the LMS in the same place in the UI as the student_courseaccessrole roles. 
+They can also be granted in the LMS Django Admin Dashboard.
+
+edx-rbac
+--------
+
+Permission is granted on a Feature. 
+
+edx-rbac is a protocol that can be implemented by any feature, but each 
+feature that implements it would need to set up its own implementation.
+
+It allows for creating feature specific roles with feature specific permissions.
+The feature specific roles can be accessed by other 
+features which can choose to use or ignore this data point. 
+
+The feature specific roles are stored on the jwt token. This mixes authz into an 
+authn data point, but is an accepted way to implement feature specific roles and permissions. 
+It is advisable to be very careful regarding the jwt token header limits if adding a new feature specific 
+set of roles using this implementation path.
+
+content_libraries_contentlibrarypermission
+------------------------------------------
+
+Permission is granted on a Feature, in this case Content Library. 
+
+Permission is assigned in the CMS exclusively for providiing explicit permission to 
+view or edit a library in the CMS. 
+
+It grants access on a library by library basis and is used for v2 of content libraries 
+in the CMS. 
+
+.. note::
+  v1 libraries (deprecated) granted access to libraries on a course by course basis 
+  and was controlled by student_courseaccessrole.
+
+student/learner
```

Comment:
[Nit]: For any document header that is not referring to something in code (e.g a repo name, method name, model name, etc.), I think it would be better to capitalize. I _think_ that is the case here, but please confirm.  Also with `system-wide roles` header below, etc.